File: src/test/java/it/marketplace/microservices/database/entity/PaymentInstallmentsEntityTest.java
package it.marketplace.microservices.database.entity;

import it.marketplace.microservices.common.enums.StatusOrderEnum;
import org.junit.jupiter.api.Test;

import java.time.LocalDateTime;

import static org.junit.jupiter.api.Assertions.*;

class PaymentInstallmentsEntityTest {

    @Test
    void shouldCreateEntityWithAllArgsConstructor_ArrangeActAssert() {
        // Arrange/Act
        LocalDateTime now = LocalDateTime.now();
        PaymentInstallmentsEntity entity = new PaymentInstallmentsEntity(1L, "REF1", "ORD1", StatusOrderEnum.PAID, 100.0, now);
        // Assert
        assertEquals(1L, entity.getId());
        assertEquals("REF1", entity.getReference());
        assertEquals("ORD1", entity.getOrderCode());
        assertEquals(StatusOrderEnum.PAID, entity.getStatus());
        assertEquals(100.0, entity.getDebit());
        assertEquals(now, entity.getTmsUpdate());
    }

    @Test
    void shouldSetAndGetFields_ArrangeActAssert() {
        // Arrange
        PaymentInstallmentsEntity entity = new PaymentInstallmentsEntity();
        LocalDateTime now = LocalDateTime.now();
        // Act
        entity.setId(2L);
        entity.setReference("REF2");
        entity.setOrderCode("ORD2");
        entity.setStatus(StatusOrderEnum.CREATED);
        entity.setDebit(200.0);
        entity.setTmsUpdate(now);
        // Assert
        assertEquals(2L, entity.getId());
        assertEquals("REF2", entity.getReference());
        assertEquals("ORD2", entity.getOrderCode());
        assertEquals(StatusOrderEnum.CREATED, entity.getStatus());
        assertEquals(200.0, entity.getDebit());
        assertEquals(now, entity.getTmsUpdate());
    }

    @Test
    void shouldTestEqualsAndHashCode_ArrangeActAssert() {
        // Arrange
        PaymentInstallmentsEntity e1 = new PaymentInstallmentsEntity();
        PaymentInstallmentsEntity e2 = new PaymentInstallmentsEntity();
        // Act
        e1.setId(1L); e2.setId(1L);
        e1.setReference("REF"); e2.setReference("REF");
        // Assert
        assertEquals(e1, e2);
        assertEquals(e1.hashCode(), e2.hashCode());
    }

    @Test
    void shouldTestToString_ArrangeActAssert() {
        // Arrange
        PaymentInstallmentsEntity entity = new PaymentInstallmentsEntity();
        entity.setReference("REF");
        // Act
        String str = entity.toString();
        // Assert
        assertTrue(str.contains("REF"));
    }
}

File: adapter/output/database/src/test/kotlin/it.marketplace/entity/PaymentInstallmentsEntityTest.kt
package it.marketplace.database.entity

import it.marketplace.common.enums.StatusOrderEnum
import it.marketplace.utils.BaseTest
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import java.time.LocalDateTime
import kotlin.test.assertEquals

class PaymentInstallmentsEntityTest : BaseTest() {

    @Test
    fun shouldCreateEntityWithAllArgsConstructor_ArrangeActAssert() {
        // Arrange/Act
        val now = LocalDateTime.now()
        val entity = mockPaymentInstallmentsEntity()
        // Assert
        assertEquals(1L, entity.id)
        assertEquals("REF1", entity.reference)
        assertEquals("ORD1", entity.orderCode)
        assertEquals(StatusOrderEnum.PAID, entity.status)
        assertEquals(100.0, entity.debit)
        assertEquals(now, entity.tmsUpdate)
    }

    @Test
    fun shouldSetAndGetFields_ArrangeActAssert() {
        // Arrange
        val now = LocalDateTime.now()
        val entity = mockPaymentInstallmentsEntity()
        // Act
        entity.reference = "REF2"
        entity.orderCode = "ORD2"
        entity.status = StatusOrderEnum.CREATED
        entity.debit = 200.0
        entity.tmsUpdate = now
        // Assert
        assertEquals("REF2", entity.reference)
        assertEquals("ORD2", entity.orderCode)
        assertEquals(StatusOrderEnum.CREATED, entity.status)
        assertEquals(200.0, entity.debit)
        assertEquals(now, entity.tmsUpdate)
    }

}

